### PR TITLE
#52 Set data-dtype for hp.temp and hp.tempmax to Number

### LINKED ISSUE #52: Set hp.temp to data-dtype Number to fix string concatenation bug
https://github.com/ElfFriend-DnD/foundryvtt-compactBeyond5eSheet/blob/4d7957cd8368f52365d8679d788ff6acbdf04901/src/templates/parts/sheet-header.hbs#L111

Temp HP isn't typed at the moment, so the value is stored as a string when using this character sheet. This leads to problems when using modules like Health Estimate, which (depending on settings) try to add Temp HP and Base HP together.

E.g. a character with 1 temp HP and 16/32 HP would show as having 116/32 HP.

image

This happens on Foundry 0.7.10, using version 0.15.0 of Compact D&D Beyond Sheet.

### PR DESCRIPTION
**Pull request by CeaneC**
Fixes #52

### CODE REVIEW

File: src/templates/parts/sheet-header.hbs [hunk 1]
```diff
@@ -108,9 +108,9 @@
             placeholder="10" />
         </div>
         <footer class="attribute-footer">
-          <input name="data.attributes.hp.temp" type="text" class="temphp" placeholder="+{{ localize 'DND5E.Temp' }}"
+          <input name="data.attributes.hp.temp" type="text" class="temphp" placeholder="+{{ localize 'DND5E.Temp' }}" data-dtype="Number"
             value="{{data.attributes.hp.temp}}" data-dtype="{{data.attributes.hp.type}}" />
```
Comment (akrigline) [inline src/templates/parts/sheet-header.hbs:112]: I notice that the line below this sets `data-dtype` to `data.attributes.hp.type`. It looks like this is how the core system did this, but that changed 7 months ago. Can you remove this old declaration as part of this? Both inputs have it.

https://gitlab.com/foundrynet/dnd5e/-/commit/d03a2b51094fd629832f1fc39f17e729892f46da

**Review by akrigline — CHANGES_REQUESTED** [review]

**Comment by akrigline** [comment]
Thanks for finding this and opening a PR! A quick cleanup of the old-bad-way and this is good to go.

File: src/templates/parts/sheet-header.hbs (hunk 1 above)
Comment (CeaneC) [reply src/templates/parts/sheet-header.hbs]: Thanks for catching that, in my hastiness to fix the bug I didn't notice the second line

**Comment by akrigline** [comment]
This completely slipped my mind... Sorry about that.

**Pull request merged**